File: catkin_ws/src/beginner_tutorials/scripts/message.py
#!/usr/bin/env python
import rospy
from beginner_tutorials.msg import Num

def talker():
    rospy.init_node('message_talker',anonymous=True)
    pub = rospy.Publisher('talker',Num)
    r = rospy.Rate(10)
    msg = Num()
    msg.num = 4
    
    while not rospy.is_shutdown():
        rospy.loginfo(msg)
        pub.publish(msg)
        r.sleep()

if __name__ == '__main__':
    try:
        talker()
    except rospy.ROSInterruptException: pass

File: catkin_ws/src/beginner_tutorials/src/message.cpp
#include "ros/ros.h"
#include "beginner_tutorials/Num.h"

int main(int argc, char **argv)
{
    ros::init(argc,argv,"talker");
    ros::NodeHandle n;
    ros::Publisher pub = n.advertise<beginner_tutorials::Num>("talker",1);
    
    ros::Rate loop_rate(10);
    
    while(ros::ok())
    {
        beginner_tutorials::Num msg;
        msg.num = 4;
        ROS_INFO("%ld",msg.num);
        pub.publish(msg);
        
        ros::spinOnce();
        loop_rate.sleep();
    }
    
    return 0;
}
